### PR TITLE
update scripts and new article

### PR DESCRIPTION
**Pull request by matstar2**
Changes to "What's New on CPAN" related-scripts:
- PAUSE id's get added to articles
- Personalization and workflow preferences
- Additional patterns for module categorization
What's New on CPAN for April 2024

### CODE REVIEW

**Comment by oalders** [comment]
Looks like the actual article URL is https://www-perl-com-pr-365.onrender.com/article/what-s-new-on-cpan-april-2024/

We should fix the render.com build not to hardcode the hostname in the urls.

File: bin/format-cpan-uploads [hunk 1]
```diff
@@ -74,59 +74,38 @@ for my $release (sort @lines)
...
     }
-    else
-    {
+    else {
       $module{category} = 'Other';
     }
 
     # confirm category or exclude
-    print "$i/$total $module{name} - accept category or select another:\n";
+    #print "$i/$total $module{name} - accept category or select another:\n";
```
Comment (oalders) [inline bin/format-cpan-uploads:106]: Delete?

File: bin/format-cpan-uploads [hunk 2]
```diff
@@ -142,14 +121,13 @@ open my $md_fh, '>:utf8', sprintf 'what-s-new-on-cpan---%s-%s.md',
...
+  tags        => ['new'],
   categories  => 'cpan',
-  image       => "",
-  draft       => JSON::true,
+  draft       => JSON::false,
   date        => $now->datetime,
-  image       => undef,
-  thumbnail   => undef,
+  image       => "/images/176/2A6DE1D0-0ACE-11E5-A57F-EAC87F6D3C83.png",
+  thumbnail   => "/images/176/thumb_2A6DE1D0-0ACE-11E5-A57F-EAC87F6D3C83.png",
```
Comment (oalders) [inline bin/format-cpan-uploads:130]: 👍🏻 It looks like David was switching between colours in his posts, but I think this is good.

File: bin/get-cpan-uploads [hunk 3]
```diff
@@ -7,6 +7,7 @@ use Getopt::Long 'GetOptions';
 use HTTP::Tiny;
 use JSON::XS 'decode_json';
 use Time::Piece;
+use Data::Dump qw/dump/;
```
Comment (oalders) [inline bin/get-cpan-uploads:10]: Since this is unused it can be deleted.

File: bin/get-cpan-uploads [hunk 4]
```diff
@@ -71,8 +72,10 @@ foreach (@upload) {
       . '|' . ($release->{date} || '')
       . '|' . ($release->{status} || '')
       . '|' . ($release->{metadata}{author}[0] || '')
+      . '|' . ($author || '')
   );
   say $line =~ s/\n|\r//gr;
+  #say STDERR dump $release->{metadata}{author};
```
Comment (oalders) [inline bin/get-cpan-uploads:78]: This can be deleted.

File: content/article/what-s-new-on-cpan---april-2024.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+	* [SortKey](https://metacpan.org/pod/SortKey) with [App::sort_by_sortkey](https://metacpan.org/pod/App::sort_by_sortkey) (PERLANCAR)
+* Access the [OpenExchangeRates](https://openexchangerates.org/) API with [Net::Async::OpenExchRates](https://metacpan.org/pod/Net::Async::OpenExchRates) (VNEALV)
+* [AI::Ollama::Client](https://metacpan.org/pod/AI::Ollama::Client) (CORION) interfaces with [Ollama](https://github.com/ollama/ollama) language models
+
+
+Config & Devops
+---------------
+* Find and build Microsoft's [ONNX Runtime](https://github.com/microsoft/onnxruntime) AI system with [Alien::onnxruntime](https://metacpan.org/pod/Alien::onnxruntime) (EGOR)
+* [CPAN::Requirements::Dynamic](https://metacpan.org/pod/CPAN::Requirements::Dynamic) (LEONT) will check that your system meets the prerequisites specified by a CPAN distribution
+* [Dist::Build](https://metacpan.org/pod/Dist::Build) (LEONT) provides another way to build and install Perl modules* [Dist::Build](https://metacpan.org/pod/Dist::Build) (LEONT) provides another way to build and install Perl modules
```
Comment (oalders) [inline content/article/what-s-new-on-cpan---april-2024.md:35]:
```suggestion
* [Dist::Build](https://metacpan.org/pod/Dist::Build) (LEONT) provides another way to build and install Perl modules
```

File: content/article/what-s-new-on-cpan---april-2024.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+Development & Version Control
+-----------------------------
+* Benchmark various ways to handle exceptions with [Bencher::Scenario::ExceptionHandling](https://metacpan.org/pod/Bencher::Scenario::ExceptionHandling) (PERLANCAR)
+* [Data::MiniDumpX](https://metacpan.org/pod/Data::MiniDumpX) (PERLANCAR) supports [Plugin::System](https://metacpan.org/pod/Plugin::System)
+* Generate fixtures for [DBD::Mock::Session](https://metacpan.org/pod/DBD::Mock::Session) with [DBD::Mock::Session::GenerateFixtures](https://metacpan.org/pod/DBD::Mock::Session::GenerateFixtures) (UXYZAB)
+* [Net::EPP::Server](https://metacpan.org/pod/Net::EPP::Server) (GBROWN) provides a high-level framework for developing EPP (Extensible Provisioning Protocol) servers
+* Load an [i3](https://i3wm.org/) workspace with [AnyEvent::I3X::Workspace::OnDemand](https://metacpan.org/pod/AnyEvent::I3X::Workspace::OnDemand) (WATERKIP)
+* Parse and format datetimes from PDF's using [DateTime::Format::PDF](https://metacpan.org/pod/DateTime::Format::PDF) (SKIM)
+* Stingify references in:
```
Comment (oalders) [inline content/article/what-s-new-on-cpan---april-2024.md:75]:
```suggestion
* Stringify references in:
```

**Review by oalders — CHANGES_REQUESTED** [review]
Looks great! Just a few comments to look at.

File: bin/format-cpan-uploads [hunk 7]
```diff
@@ -11,7 +11,7 @@ use Time::Seconds;
 
 # dont leave tmp file behind
 BEGIN { $SIG{TERM} = $SIG{INT} = sub { exit 0 } }
-END { unlink '.tmp.pod' }
+#END { unlink '.tmp.pod' } #don't need to cleanup since we're not using this file
```
[the remark this answers is not in the document]
Comment (oalders) [reply bin/format-cpan-uploads:14]: > It also occurs to me that I should have further subdivided those script changes into commits of their own.

That's always helpful. 😄 They could even live in their own PR, but I wouldn't worry too much about it today. It's nice just to see activity on this repository.

File: bin/get-cpan-uploads (hunk 4 above)
Comment (matstar2) [reply bin/get-cpan-uploads]: Done

File: bin/get-cpan-uploads (hunk 3 above)
Comment (matstar2) [reply bin/get-cpan-uploads]: Done

File: bin/format-cpan-uploads (hunk 1 above)
Comment (matstar2) [reply bin/format-cpan-uploads]: Done

File: bin/format-cpan-uploads [hunk 8]
```diff
@@ -56,14 +56,14 @@ for my $release (sort @lines)
...
-    open my $tmp, '>:utf8', '.tmp.pod';
-    print $tmp $response->{content};
-    close $tmp; # flush it
-    eval { system "perldoc .tmp.pod" };
-    next if $@;
+    #open my $tmp, '>:utf8', '.tmp.pod';
+    #print $tmp $response->{content};
+    #close $tmp; # flush it
+    #eval { system "perldoc .tmp.pod" };
+    #next if $@;
```
[the remark this answers is not in the document]
Comment (matstar2) [reply bin/format-cpan-uploads]: Done

File: bin/format-cpan-uploads (hunk 7 above)
[the remark this answers is not in the document]
Comment (matstar2) [reply bin/format-cpan-uploads]: Done

**Comment by matstar2** [comment]
I believe this is now ready to go

**Review by oalders — APPROVED** [review]
🎉

**Comment by mohawk2** [comment]
I had a read of the March and April articles. I appreciate that it's curated and therefore this may have been a conscious choice, but while there was lots of PERLANCAR modules, there were no PDL modules shown (there was a release of actual PDL in both months, along with various other modules). Is that deliberate?

**Comment by oalders** [comment]
The What's New list is a list of first ever releases for modules. Were the PDL releases being uploaded for the very first time in that namespace?

It might be helpful to include noteworthy modules that have also been released but that's not traditionally with this article has been.

**Comment by mohawk2** [comment]
Ah, sorry for my confusion - I read "what's new" as including "a new version", not "completely new distro". It might make sense to emphasise that in some way to help the hard of understanding :-)

**Pull request merged**